zingolib/src/lightclient/send.rs

```diff
@@ -30,11 +66,44 @@ impl LightClient {
     #[cfg(feature = "zip317")]
     pub async fn do_propose_spend(
         &self,
-        _receivers: Vec<(Address, NonNegativeAmount, Option<MemoBytes>)>,
+        receivers: Vec<(Address, NonNegativeAmount, Option<MemoBytes>)>,
     ) -> Result<crate::data::proposal::TransferProposal, String> {
-        use crate::test_framework::mocks::ProposalBuilder;
+        let request =
+            receivers_becomes_transaction_request(receivers).map_err(|e| e.to_string())?;
+
+        let change_strategy = zcash_client_backend::fees::zip317::SingleOutputChangeStrategy::new(
+            zcash_primitives::transaction::fees::zip317::FeeRule::standard(),
+            None,
+            ShieldedProtocol::Orchard,
+        ); // review consider change strategy!
+
+        let input_selector = GISKit::new(
+            change_strategy,
+            zcash_client_backend::fees::DustOutputPolicy::default(),
+        );
+
+        let mut tmamt = self
+            .wallet
+            .transaction_context
+            .transaction_metadata_set
+            .write()
+            .await;
+
+        let proposal = zcash_client_backend::data_api::wallet::propose_transfer::<
+            TxMapAndMaybeTrees,
+            ChainType,
+            GISKit,
+            ZingoLibError,
+        >(
+            tmamt.deref_mut(),
+            &self.wallet.transaction_context.config.chain,
+            zcash_primitives::zip32::AccountId::ZERO,
+            &input_selector,
+            request,
+            NonZeroU32::new(1).expect("yeep yop"), //review! be more specific
```

Comment:
```suggestion
            NonZeroU32::MIN
```
Would do this without the expect. For non-one numbers, here's also the possibility of moving the panic into a helper `const fn` and that way would only be hit at compile time